src/pkg/utils/sbom.go

```diff
@@ -86,3 +86,32 @@ func SBOMExtractor(dst string, SBOMArtifactPathMap map[string]string) func(_ con
 	}
 	return extractor
 }
+
+// HandleSBOM handles the extraction and creation of bundle SBOMs after populating SBOMArtifactPathMap
+func HandleSBOM(extractSBOM bool, SBOMArtifactPathMap map[string]string, bundleName, dstPath string) ([]string, error) {
```

Comment:
Nit: We should probably unit test this function since it's exported. 

Is there any case where `HandleSBOM` will return more than 1 warning? Reading the code I only see 2 `append` calls for the slice, and they both end up returning immediately after, so they wouldn't both be called in a single invocation.

Reply:
I'm wondering if we will ever return a warning and an error? Right now the code doesn't look like it, but maybe we anticipate things in the future?

Reply:
> Nit: We should probably unit test this function since it's exported.

IMO we've got this pretty well covered in our e2e suite. Unit testing would be difficult because of the reliance on the tarball lib

Reply:
reviewed sync, made a couple refactors